intro-unal-2022-1-tanks/Assets/Scr/TurretMovement.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TurretMovement : MonoBehaviour
+{
+    [SerializeField] private Transform playerTank;
+    [SerializeField] private Transform[] turretCanyon;
+    [SerializeField] private Transform maxY;
+    [SerializeField] private Transform minY;
+    public float speed = 0.4f;
+
+    void Start()
+    {
+
+    }
+
+    void Update()
+    {
+        transform.position = Vector3.Lerp(minY.position, maxY.position, Mathf.PingPong(Time.time * speed, 1.0f));
+        
+        Vector3 direction = playerTank.position - transform.position;
```

Comment:
Perfecto!
En caso de que se quiera más presición, este cálculo se deberia hacer dentro del for y usando el transform de cada torreta en vez del transform del padre. Así tomas en cuenta ese desfase que tiene cada torreta respecto al (0, 0) del padre.